pkg/api/node_api.go

```diff
@@ -649,31 +649,50 @@ func (a *NodeApi) unconfirmedSize(w http.ResponseWriter, _ *http.Request) error
 	return nil
 }
 
-type rollbackRequest struct {
-	Height uint64 `json:"height"`
+type rollbackResponse struct {
+	BlockID proto.BlockID `json:"blockId"`
 }
 
-type rollbackToHeight interface {
-	RollbackToHeight(string, proto.Height) error
+func (a *NodeApi) RollbackToHeight(w http.ResponseWriter, r *http.Request) error {
+	type rollbackRequest struct {
+		Height                  uint64 `json:"rollbackTo"`
+		ReturnTransactionsToUtx bool   `json:"returnTransactionsToUtx"`
+	}
+
+	rollbackReq := &rollbackRequest{}
+	if err := tryParseJson(r.Body, rollbackReq); err != nil {
+		return errors.Wrap(err, "failed to parse RollbackToHeight body as JSON")
+	}
+	err := a.state.RollbackToHeight(rollbackReq.Height)
```

Comment:
Consider error handling like at `pkg/api/node_api.go:95`. Function `state.RollbackToHeight` can return `RetrievalError` which probably should be considered as 404 HTTP error.

Reply:
Note that `RollbackTo` function doesn't return RetrievalError.